### PR TITLE
Use ddata mode as the default for Cluster Sharding, #17963

### PR DESCRIPTION
**Pull request by patriknw**

### CODE REVIEW

**Comment by patriknw** [comment]
Refs #17963

File: akka-cluster-sharding/src/main/resources/reference.conf [hunk 1]
```diff
@@ -58,22 +58,24 @@ akka.cluster.sharding {
   # used for the internal persistence of ClusterSharding. If not defined
   # the default journal plugin is used. Note that this is not related to
   # persistence used by the entity actors.
+  # Only used when state-store-mode=persistence
```
Comment (ktoso) [inline akka-cluster-sharding/src/main/resources/reference.conf]: Good

File: akka-cluster-sharding/src/main/resources/reference.conf [hunk 2]
```diff
@@ -58,22 +58,24 @@ akka.cluster.sharding {
...
 
-  # Parameter which determines how the coordinator will be store a state
-  # valid values either "persistence" or "ddata"
-  # The "ddata" mode is experimental, since it depends on the experimental
-  # module akka-distributed-data-experimental.
-  state-store-mode = "persistence"
+  # Defines how the coordinator stores its state. Same is also used by the
+  # shards for rememberEntities.
+  # Valid values are "ddata" or "persistence". 
+  state-store-mode = "ddata"
```
Comment (ktoso) [inline akka-cluster-sharding/src/main/resources/reference.conf]: Yay!

File: akka-docs/rst/java/cluster-sharding.rst [hunk 3]
```diff
@@ -208,14 +225,18 @@ The settings for Distributed Data is configured in the the section
...
+^^^^^^^^^^^^^^^^
 
-.. warning::
+This mode is enabled with configuration::
+
+  akka.cluster.sharding.state-store-mode = persistence
+
+Since it is running in a cluster :ref:`persistence-java` must be configured with a distributed journal.
+
+You must explicitly add the ``akka-persistence`` dependency to your build if
```
Comment (ktoso) [inline akka-docs/rst/java/cluster-sharding.rst]: Add some note that it is very important to watch out for cluster splits and not use auto downing with it? Just a small sentence would be enough o think, no need for a note block

**Review by ktoso — APPROVED** [review]
LGTM, very cool

File: akka-docs/rst/java/cluster-sharding.rst (hunk 3 above)
Comment (patriknw) [inline akka-docs/rst/java/cluster-sharding.rst]: That is independent of the choice of `state-store-mode`. Equally important with `ddata`. We already have a red box at the [top of the page](http://doc.akka.io/docs/akka/2.4/scala/cluster-sharding.html).

Comment (ktoso) [inline akka-docs/rst/java/cluster-sharding.rst]: Ok, cool.

**Pull request merged**